### PR TITLE
add option to run local operators

### PR DESCRIPTION
**Pull request by maelk**
Instead of using 'make deploy' for Baremetal Operator and CAPI
provider baremetal, add an option to use 'make run' and log the
output.

### CODE REVIEW

**Comment by nordixinfra** [comment]
Can one of the admins verify this patch?

File: 03_launch_mgmt_cluster.sh [hunk 1]
```diff
@@ -7,6 +7,7 @@ source lib/logging.sh
 source lib/common.sh
 
 eval "$(go env)"
+export GOPATH
```
Comment (kanwar-saad) [inline 03_launch_mgmt_cluster.sh:10]: a small header for the script would be nice .. now i dont understand in the first glance that what are the inputs for this script.

File: 03_launch_mgmt_cluster.sh [hunk 2]
```diff
@@ -63,7 +67,15 @@ function launch_minikube() {
 
 function launch_baremetal_operator() {
     pushd "${BMOPATH}"
-    make deploy
+    if [ "${BMO_RUN_LOCAL}" = true ]; then
+      touch bmo.out.log
+      touch bmo.err.log
+      make deploy
+      kubectl scale deployment metal3-baremetal-operator -n metal3 --replicas=0
```
Comment (kanwar-saad) [inline 03_launch_mgmt_cluster.sh:88]: can we just delete those pods or deployment?

File: 04_verify.sh [hunk 3]
```diff
@@ -105,9 +105,12 @@ check_k8s_entity() {
...
     # Check the replicas
-    FAILS="$(equals "$(echo "${ENTITY}" | jq -r '.status.readyReplicas')" \
-      "$(echo "${ENTITY}" | jq -r '.status.replicas')" \
-      "${name} ${1} replicas correct")"
+    if [[ "${BMO_RUN_LOCAL}" != true ]] && [[ "${CAPBM_RUN_LOCAL}" != true ]]
+    then
+      FAILS="$(equals "$(echo "${ENTITY}" | jq -r '.status.readyReplicas')" \
+        "$(echo "${ENTITY}" | jq -r '.status.replicas')" \
+        "${name} ${1} replicas correct")"
+    fi
```
Comment (kanwar-saad) [inline 04_verify.sh:113]: shouldn't there be an else checking that local process is running?

**Pull request merged**